### PR TITLE
Structure_Engine: Rename StartNode/EndNode to Start/End

### PR DESCRIPTION
**Pull request by FraserGreenroyd**
### NOTE: Depends on 
https://github.com/BHoM/BHoM/pull/1579

### Changelog
 - Update Analytical and Structure Engine based on property rename from `StartNode` to `Start` and from `EndNode` to `End`

### CODE REVIEW

**Comment by FraserGreenroyd** [comment]
@BHoMBot check installer

File: Structure_Engine/Query/IsNull.cs [hunk 1]
```diff
@@ -83,7 +83,7 @@ public static bool IsNull(this Bar bar, string msg = "", [CallerMemberName] stri
                 ErrorMessage(methodName, "Bar", msg);
                 return true;
             }
-            else if (bar.StartNode.IsNull("The Node (StartNode) is owned by a Bar.", methodName) || bar.EndNode.IsNull("The Node (EndNode) is owned by a Bar.", methodName))
+            else if (bar.Start.IsNull("The Node (StartNode) is owned by a Bar.", methodName) || bar.End.IsNull("The Node (EndNode) is owned by a Bar.", methodName))
```
Comment (FraserGreenroyd) [inline Structure_Engine/Query/IsNull.cs:86]:
```suggestion
            else if (bar.Start.IsNull("The Node (Start) is owned by a Bar.", methodName) || bar.End.IsNull("The Node (End) is owned by a Bar.", methodName))
```

**Comment by FraserGreenroyd** [comment]
@BHoMBot check installer

**Comment by FraserGreenroyd** [comment]
@BHoMBot check installer

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance
@BHoMBot check core
@BHoMBot check null-handling
@BHoMBot check serialisation

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check installer

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core
@BHoMBot check null-handling
@BHoMBot check serialisation

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am requesting neutral checks on: unit-tests

**Review by Tom-Kingstone — APPROVED** [review]
Approved, see message in linked PR.

**Pull request merged**